### PR TITLE
feat(banners): add new banner variants - FRONT-1153

### PR DESCRIPTION
**Pull request by emeryro**
# PR description

specs: 
- https://citnet.tech.ec.europa.eu/CITnet/confluence/display/NEXTEUROPA/Page+banner++-++Design+specifications
- https://citnet.tech.ec.europa.eu/CITnet/confluence/display/NEXTEUROPA/Hero+banner++-++Design+specifications

breaking changes:
- banners markup had to be slightly updated (mostly separating two containers)
- `baseline` has been renamed to `description` in page banner, as it is the term used in the documentation. Hero banner was already using this correctly.
- some variants had to be renamed

### CODE REVIEW

File: src/systems/ec/implementations/react/components/page-banner/src/PageBanner.jsx [hunk 1]
```diff
@@ -19,31 +21,37 @@ const PageBanner = ({
...
+            )}
+            {link && link.label && (
+              <Link
+                {...link}
+                variant="cta"
+                className="ecl-page-banner__link-cta"
+              />
+            )}
+            {/* DEPRECATED */}
+            {button && button.label && (
```
Comment (kalinchernev) [inline src/systems/ec/implementations/react/components/page-banner/src/PageBanner.jsx]: The pull request already contains breaking changes. Why keeping something which is deprecated?

File: src/systems/ec/implementations/react/components/page-banner/stories/Index.jsx [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
-import React from 'react';
+import React, { Fragment } from 'react';
 import { withKnobs, text, boolean } from '@storybook/addon-knobs';
 
 import demoContentImage from '@ecl/ec-specs-page-banner/demo/data--image';
 import demoContentImageShade from '@ecl/ec-specs-page-banner/demo/data--image-shade';
-import demoContentPrimary from '@ecl/ec-specs-page-banner/demo/data--primary';
-import demoContentDefault from '@ecl/ec-specs-page-banner/demo/data--default';
-import demoContentAlignLeft from '@ecl/ec-specs-page-banner/demo/data--align-left';
+import demoContentSimple from '@ecl/ec-specs-page-banner/demo/data--simple';
```
Comment (kalinchernev) [inline src/systems/ec/implementations/react/components/page-banner/stories/Index.jsx]: I'd consider this change in spec structure a breaking change too in the ecl-twig consumer context.

File: src/systems/ec/implementations/react/components/page-banner/stories/Index.jsx [hunk 3]
```diff
@@ -15,109 +13,502 @@ export default {
   decorators: [withKnobs],
 };
 
+/*
+ * Image
```
Comment (kalinchernev) [inline src/systems/ec/implementations/react/components/page-banner/stories/Index.jsx]: The file has grown to be harder to maintain and it's understandable to have these comments.
My suggestion would be to split stories into separate files and import+present in this index file

File: src/systems/ec/implementations/react/components/page-banner/stories/Index.jsx [hunk 4]
```diff
@@ -15,109 +13,502 @@ export default {
...
+    true,
+    'Banner display'
+  );
+  const fullWidthGrid = boolean(
+    'Full width (inside the grid)',
+    false,
+    'Banner display'
+  );
 
+  if (fullWidth) {
```
Comment (kalinchernev) [inline src/systems/ec/implementations/react/components/page-banner/stories/Index.jsx]:
```jsx
  const pageBanner = (
    <PageBanner
      {...demoContentImage}
      variant="image"
      title={title}
      description={description}
      link={link}
      isCentered={centered}
      isFullWidth={fullWidthGrid}
      image={image}
    />
  );

  return fullWidth ? (
    pageBanner
  ) : (
    <div className="ecl-container">{pageBanner}</div>
  );
```

File: src/systems/ec/implementations/react/components/page-banner/stories/Index.jsx [hunk 5]
```diff
@@ -15,109 +13,502 @@ export default {
...
+      <Fragment>
+        <p className="ecl-u-type-paragraph">
+          Note: this variant comes with a transparent background; the image here
+          in inline style is just for demo
+        </p>
+        <PageBanner
+          style={{
+            'background-image':
+              "url('https://inno-ecl.s3.amazonaws.com/media/examples/example-image11.jpg')",
+            'background-size': 'cover',
```
Comment (kalinchernev) [inline src/systems/ec/implementations/react/components/page-banner/stories/Index.jsx]: React has few suggestions on these properties

screenshot-nimbus-capture-2020 07 09-10_52_03

I checked that the component itself has the pattern already, will be nice to update here as well

File: src/systems/eu/implementations/react/components/hero-banner/stories/Index.jsx [hunk 6]
```diff
@@ -1,97 +1,514 @@
...
-import React from 'react';
-import { storiesOf } from '@storybook/react';
+import React, { Fragment } from 'react';
 import { withKnobs, text, boolean } from '@storybook/addon-knobs';
 
 import demoContentImage from '@ecl/eu-specs-hero-banner/demo/data--image';
 import demoContentImageShade from '@ecl/eu-specs-hero-banner/demo/data--image-shade';
-import demoContentPrimary from '@ecl/eu-specs-hero-banner/demo/data--primary';
-import demoContentDefault from '@ecl/eu-specs-hero-banner/demo/data--default';
-import demoContentAlignLeft from '@ecl/eu-specs-hero-banner/demo/data--align-left';
```
Comment (kalinchernev) [inline src/systems/eu/implementations/react/components/hero-banner/stories/Index.jsx:10]: Same suggestion for breaking change note on the change of data structure (spec)

File: src/systems/ec/implementations/vanilla/packages/ec-component-hero-banner/ec-component-hero-banner.scss [hunk 7]
```diff
@@ -12,97 +12,140 @@
 );
 
 @mixin ecl-hero-banner(
-  $background-color-default: $ecl-color-grey-5,
```
Comment (kalinchernev) [inline src/systems/ec/implementations/vanilla/packages/ec-component-hero-banner/ec-component-hero-banner.scss:15]: Just wondering, wasn't it the case that this approach of passing variables for theming has been deprecated?

**Review by kalinchernev — COMMENTED** [review]
No issues found, only leaving a few comments after reading through the changes

**Review by kalinchernev — DISMISSED** [review]
Giving a green light as technically no issues apart from the suggestions

File: src/systems/ec/implementations/react/components/page-banner/stories/Index.jsx (hunk 3 above)
Comment (emeryro) [reply src/systems/ec/implementations/react/components/page-banner/stories/Index.jsx]: Do you have an example of split stories like that? I didn't succeed in making it work...

File: src/systems/ec/implementations/vanilla/packages/ec-component-hero-banner/ec-component-hero-banner.scss (hunk 7 above)
Comment (emeryro) [reply src/systems/ec/implementations/vanilla/packages/ec-component-hero-banner/ec-component-hero-banner.scss:15]: there will most probably not be other themes in ECL V2. That will be for next major version, and then we will indeed have to make sure to use correct variables

File: src/systems/ec/implementations/react/components/page-banner/stories/Index.jsx (hunk 3 above)
Comment (kalinchernev) [reply src/systems/ec/implementations/react/components/page-banner/stories/Index.jsx]: Opened a pull request with comparison to this one here https://github.com/ec-europa/europa-component-library/pull/1634

**Review by kalinchernev — APPROVED** [review]

**Pull request merged**